packages/babel-traverse/src/index.js

```diff
@@ -33,6 +33,7 @@ traverse.explode = visitors.explode;
 traverse.NodePath = require("./path");
 traverse.Scope    = require("./scope");
 traverse.Hub      = require("./hub");
+traverse.cache    = require("./cache");
```

Comment:
To your point for

> .clear() can be called on the cache by a plugin to clear
potentially outdated information. This is helpful when implementing two
seperate pipelines that should not share information.

Are you thinking then of two independent passes over the AST by Babel on the same AST? As a standalone API, I worry exposing all of `cache` will be hard to maintain, since a WeakMap is an extremely broad implementation. If clearing is the goal, perhaps exposing only that part would be more maintainable?